### ISSUE TITLE
Fail test if `Before` section fails

### ISSUE THREAD

**Issue by NivLipetz** [Zooz/predator#201]
**Describe the solution you'd like**
If a request in the `Before` flow fails, I would like the test to fail. This can be done by adding for each request in the before flow an optional parameter such as `allow_failure: false` while the default value would be `true`. If a request with the `allow_failure: false` fails, then fail the test; otherwise proceed.

**Comment by enudler** [comment]
How do you define failed request? By status code? Capture? Assert?

**Comment by NivLipetz** [comment]
Good point. I suggest we use `artillery-plugin-expect` documented here: https://artillery.io/docs/plugin-expectations-assertions/

![image](https://user-images.githubusercontent.com/31369413/62856061-8aebcb00-bcfc-11e9-9758-e95cf408c7b7.png)

**Comment by enudler** [comment]
Sounds good or we can use asserts which are supported out of the box in artillery 
I will add it to 1.2.0

**Comment by enudler** [comment]
It can be also implemented using the processors feature:

For example:
`
function throwErrorOnError(requestParams, response, context, ee, next) {
    if (response.statusCode !== 200 || response.statusCode !== 201 || response.statusCode !== 204){
        throw new Error('Stopping test');
    }
}`

We will need to catch this errors and notify predator that test was aborted.
Adding it to 1.3.0

**Comment by enudler** [comment]
I suggest aborting the test if a capture fails in the 'before' section
Change need to be done in predator-runner
